lib/matplotlib/backend_managers.py

```diff
@@ -14,6 +18,197 @@
+from matplotlib.figure import Figure
+from matplotlib.backends import get_backend, backend as backend_name
+
+
+class FigureManagerEvent(object):
+    """Event for when something happens to this figure manager.
+    i.e. the figure it controls gets closed
+
+    Attributes
+    ----------
+    signal : str
+        The name of the signal.
+
+    figure_manager : FigureManager
+        The figure manager that fired the event.
+    """
+    def __init__(self, signal, figure_manager):
+        self.name = signal
+        self.figure_manager = figure_manager
+
+
+class FigureManager(cbook.EventEmitter):
+    """
+    The FigureManager creates and wraps the necessary components to display a
+    figure, namely the Window, FigureCanvas and Toolbar.  It gets used whenever
+    you want the figure in a standalone window.
+
+    Parameters
+    ----------
+    figure : `matplotlib.figure.Figure`
+        The figure to manage.
+
+    num : int
+        The figure number.
+
+    Attributes
+    ----------
+
+    canvas : `matplotlib.backend_bases.FigureCanvasBase`
+        The GUI element on which we draw.
+
+    figure : `matplotlib.figure.Figure`
+        The figure that holds the canvas
+
+    toolbar : `matplotlib.backend_bases.ToolbarBase`
+        The toolbar used for interacting with the figure.
+
+    window : `matplotlib.backend_bases.WindowBase`
+        The window that holds the canvas and toolbar.
+
+    num : int
+        The figure number.
+    """
+    def __init__(self, figure, num, **kwargs):
+        super(FigureManager, self).__init__(**kwargs)
+        self._backend_name, self._backend = get_backend()
+
+        self.num = num
+        self.figure = figure
+
+        self._is_gui = hasattr(self._backend, 'Window')
+        if not self._is_gui:
+            self.window = None
+            return
+
+        self._mainloop = self._backend.MainLoop()
+        self.window = self._backend.Window('Figure %d' % num)
+        self.window.mpl_connect('window_destroy_event', self.destroy)
+
+        w = int(self.figure.bbox.width)
+        h = int(self.figure.bbox.height)
+
+        self.window.add_element(self.figure.canvas, 'center')
+
+        self.toolmanager = ToolManager(self.figure)
+        self.toolbar = self._get_toolbar()
+
+        tools.add_tools_to_manager(self.toolmanager)
+        if self.toolbar:
+            tools.add_tools_to_container(self.toolbar)
+            self.statusbar = self._backend.Statusbar(self.toolmanager)
+            h += self.window.add_element(self.statusbar, 'south')
+
+        if self.toolbar is not None:
+            h += self.window.add_element(self.toolbar, 'south')
+
+        self.window.set_default_size(w, h)
+        self._full_screen_flag = False
+
+        if is_interactive():
+            self.window.show()
+
+        def notify_axes_change(fig):
+            'this will be called whenever the current axes is changed'
+            if self.toolmanager is None and self.toolbar is not None:
```

Comment:
Do we need the add_axobserver? it is already handled by toolmanager and as far as I know MEP27 is going to be used exclusively with toolmanager.